### PR TITLE
GLSP-954: Improve DI configuration

### PR DESCRIPTION
**Pull request by tortmayr**
- Reuse new di utility functions provided by @eclipse-glsp/protocol
- Ensure that diagram containers are created as proper child containers of the main Theia process => All services of the main container are accessible in the diagram container
-  Refactor DiagramConfiguration API
- Remove unused Logger binding (backend)

Also: Refactor `TheiaGLSPSelectionForwarder`
- Implement as SelectionListener instead of SelectActionHandler. This ensures that the forwarder receives all selection changes (e.g. SelectAllAction where not forwarded before)

Part of https://github.com/eclipse-glsp/glsp/issues/849 
Part of https://github.com/eclipse-glsp/glsp/issues/954

### CODE REVIEW

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Hi @tortmayr,

In general the code looks good to me but either I'm missing some kind of dependency or something else but for me, when opening a diagram I get the following injection error:
```
container.ts:154 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'options')
    at __webpack_modules__.../../node_modules/inversify/lib/container/container.js.Container.bind (container.ts:154:28)
    at bindAsService (di-util.ts:164:5)
    at WorkflowDiagramConfiguration.initializeContainer (glsp-diagram-configuration.ts:103:22)
    at WorkflowDiagramConfiguration.createContainer (glsp-diagram-configuration.ts:88:14)
    at ConfigurableGLSPDiagramManager.createWidget (glsp-diagram-manager.ts:152:40)
    at WidgetManager.getOrCreateWidget (widget-manager.ts:223:43)
    at ConfigurableGLSPDiagramManager.getOrCreateWidget (widget-open-handler.ts:151:35)
    at ConfigurableGLSPDiagramManager.open (widget-open-handler.ts:89:35)
    at open (opener-service.ts:90:19)
```

Have you seen that before?

**Comment by tortmayr** [comment]
> Hi @tortmayr,
> 
> In general the code looks good to me but either I'm missing some kind of dependency or something else but for me, when opening a diagram I get the following injection error:
> 
> ```
> container.ts:154 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'options')
>     at __webpack_modules__.../../node_modules/inversify/lib/container/container.js.Container.bind (container.ts:154:28)
>     at bindAsService (di-util.ts:164:5)
>     at WorkflowDiagramConfiguration.initializeContainer (glsp-diagram-configuration.ts:103:22)
>     at WorkflowDiagramConfiguration.createContainer (glsp-diagram-configuration.ts:88:14)
>     at ConfigurableGLSPDiagramManager.createWidget (glsp-diagram-manager.ts:152:40)
>     at WidgetManager.getOrCreateWidget (widget-manager.ts:223:43)
>     at ConfigurableGLSPDiagramManager.getOrCreateWidget (widget-open-handler.ts:151:35)
>     at ConfigurableGLSPDiagramManager.open (widget-open-handler.ts:89:35)
>     at open (opener-service.ts:90:19)
> ```
> 
> Have you seen that before?

Looks like the glsp-client version in the yarn.lock file was outdated. I pushed an update and it should work now.

File: packages/theia-integration/src/browser/theia-integration-frontend-module.ts [hunk 1]
```diff
@@ -41,44 +42,44 @@ import { TheiaOpenerOptionsNavigationService } from './theia-opener-options-navi
...
-    if (isBound(NotificationManager)) {
-        rebind(NotificationManager).toService(GLSPNotificationManager);
-    } else {
-        bind(NotificationManager).toService(GLSPNotificationManager);
-    }
+    bindOrRebind(context, NotificationManager).toService(GLSPNotificationManager);
 
     bind(GLSPDiagramContextKeyService).toSelf().inSingletonScope();
+
+    // Can be moved into diagram container right?
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply packages/theia-integration/src/browser/theia-integration-frontend-module.ts]: Removed (the comment) 😉

**Review by martin-fleck-at — APPROVED** [review]
Thank you for the quick update, LGTM!

**Pull request merged**